src/features/organizations/utils/generateTreeData.spec.ts

```diff
@@ -185,35 +99,40 @@ describe('generateTreeData()', () => {
   });
 });
 
-function mockOrganization(
-  organization: Pick<ZetkinOrganization, 'id' | 'title' | 'parent'>
-): ZetkinOrganization {
-  const injectedProperties = {
+function mockOrganization(id: number, parentId?: number): ZetkinOrganization {
+  return {
     avatar_file: null,
     country: 'SE',
     email: null,
+    id: id,
     is_active: false,
     is_open: false,
     is_public: true,
     lang: null,
-    parent: null,
+    parent: parentId
+      ? {
+          id: parentId,
+          title: `Org ${parentId}`,
+        }
+      : null,
     phone: null,
     slug: 'slug',
+    title: `Org ${id}`,
   };
-
-  return { ...organization, ...injectedProperties };
```

Comment:
On this line, whatever is in `injectedProperties` overrides whatever is in `organization`. But `organization` is the data that the user of the function (the tests) are passing in as the definition of the organization.

This means, that all organizations returned by this function will always have `parent: null`, because that's what the `injectedProperties` object has, regardless of what the user of the function passes in.

This lead to false positives, i.e. tests were passing not because the algorithm they were testing was correct, but because the tests themselves were wrong. That's always a risk when writing tests, especially when writing them _after_ the code under test, because when you write the tests first, before you implement the logic ([TDD](https://en.wikipedia.org/wiki/Test-driven_development)), you will notice if they're not failing at first.

_Originally posted by @richardolsson in https://github.com/zetkin/app.zetkin.org/pull/1410#discussion_r1214626543_